internal/resources/iampolicy/resource_iam_policy.go

```diff
@@ -0,0 +1,505 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package iampolicy
+
+import (
+	"context"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	iammodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/iam_policy"
+	clusteriammodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/iam_policy/cluster"
+	clustergroupiammodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/iam_policy/clustergroup"
+	namespaceiammodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/iam_policy/namespace"
+	organizationiammodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/iam_policy/organization"
+	workspaceiammodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/iam_policy/workspace"
+	objectmetamodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/objectmeta"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+type (
+	contextMethodKey struct{}
+)
+
+func ResourceIAMPolicy() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   resourceIAMPolicyRead,
+		CreateContext: resourceIAMPolicyCreate,
+		UpdateContext: resourceIAMPolicyInPlaceUpdate,
+		DeleteContext: resourceIAMPolicyDelete,
+		Schema:        iamPolicySchema,
+		CustomizeDiff: customdiff.All(
+			validateScope, validateRoleBindingSubjectDuplicate),
+	}
```

Comment:
`customdiff.All` is not required if only one validation exists. Instead, please add the scope block validation.